juju/juju
# fix bundle loop caused by recent change sort algorithm updates.

A loop deploying a complicated bundle was recently reported in the juju MM channel.  It turned out to be in applicationsForMachineChange.  This method relied on the old changes sort to work.  

The method was meant error reporting or debugging, therefore do not call for logging, if debug is not used.

## QA steps

The following bundle will fail to create machines, it's intended for a Maas config.  However the command should complete without error.  https://pastebin.canonical.com/p/5VsyJcDGGN/ 

It has been slightly modified from the original report, to complete without Maas spaces and tags.  It does reproduce the bug reported.

```sh
juju bootstrap localhost test 
juju deploy loop-bundle.yaml
```

Also try a bundle intended to upgrade an existing charm.
